src/components/FormInput/FormInput.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from "react";
+import "./FormInput.css";
+
+interface FormInputProps {
+  id: number,
+  value: string,
+  name: string,
+  type: string,
+  placeholder: string,
+  errorMessage: string,
+  label: string,
+  pattern?: string,
+  required: boolean,
+  onChange: (e: React.SyntheticEvent) => void
+  // onChange: React.ChangeEventHandler<HTMLInputElement>
+}
+
+const FormInput:React.FC<FormInputProps> = (props) => {
+  const [focused, setFocused] = useState(false);
+  const { label, errorMessage, onChange, id, ...inputProps } = props;
+
+  const handleFocus = (e: React.SyntheticEvent) => {
```

Comment:
```suggestion
  const handleFocus = () => {
```
Надо настроить, чтобы были варнинги на неиспользуемое, чтобы оно не проникало в код.